### ISSUE TITLE
Comparison table

### ISSUE THREAD

**Issue by 1440000bytes** [taproot-wizards/purrfect_vault#1]
How does this vault implementation differ from vaults created using OP_CTV, OP_VAULT and other opcodes?

A table comparing basic things added in README would be helpful.

**Comment by Psifour** [comment]
I just saw this now, you are correct that a table would be very useful in explaining the differences and why we still would want CTV/VAULT even in a world where we have CAT. I will see if someone on the team can spare some time to do this and/or if I can make time to do it myself.

The short answer, they all are excellent tools for the specific usecase they are optimized for, but just like you have more than one tool to work on your home, you may find that you want more than one tool to work on your complex covenant needs.

**Comment by 1440000bytes** [comment]
I think one basic difference is that other vault implementations won't use so many opcodes in the transactions: https://github.com/taproot-wizards/purrfect_vault/blob/main/src/vault/script.rs

**Comment by securitybrahh** [comment]
> why we still would want CTV/VAULT even in a world where we have CAT.

op_CAT is one of the oldest op codes? if this pool succeed, I don't think we need op_CTV.

op_VAULT solves completely different problem.

**Comment by Psifour** [comment]
> ... won't use so many opcodes ...

This is correct, but it is also worth noting that this was one of the first functional examples published and we have identified many improvements to script size since then (namely that the alt stack should be avoided unless you need it's unique characteristics). Early on most of our (@rot13maxi) work used the alt stack heavily as it is typically the most expedient way to test a concept, but can increase script size and decrease readability in some cases.

> ... I don't think we need op_CTV.

We don't 'need' CTV as CAT *can* fully recreate it, but CTV is much more succinct (smaller script size) for the things it can introspect, but it lacks the ability to look into every aspect of a serialized transaction, it's parents (or even grandparents), etc.

We need more of BOTH and more documentation and resources to help explain the differences and benefits of each. Too many people are relying on ChatGPT for their hot takes on twitter and it is comically bad at understanding bitcoin and/or script.
